algorithms_task_2/student/src/impl.cpp

```diff
@@ -1 +1,6 @@
 #include "impl.h"
+
+bool startsWith(const std::string &source, const std::string &prefix)
+{
+    return source.find(prefix) != std::string::npos;
```

Comment:
has overhead, prefer algorithms version

p.s: think why it would have an overhead

Reply:
Overhead, because string::find has O(prefix.size() * source.size()) complexity, but std::equal has at most last1 - first1 applications of the predicate for C++11